code/modules/status_system/statusEffects.dm

```diff
@@ -2975,3 +2975,276 @@
 		src.art = null
 		qdel(src.glimmer)
 		src.glimmer = null
+
+/datum/statusEffect/art_curse
+	icon_state = "art_curse"
+	desc = "You've been cursed by an Eldritch artifact!"
+	unique = FALSE
+	effect_quality = STATUS_QUALITY_NEGATIVE
+
+	var/extra_desc = ""
+	var/removal_msg = ""
+	var/outputs_desc = TRUE
+	var/outputs_removal_msg = TRUE
+
+	var/datum/artifact/curser/linked_curser
+
+	New()
+		src.desc += " [src.extra_desc]"
+		..()
+
+	preCheck(atom/A)
+		. = ..()
+		if (!ishuman(A))
+			return FALSE
+
+	onAdd(optional)
+		..()
+		if (src.outputs_desc)
+			boutput(src.owner, SPAN_ALERT(src.desc))
+		src.linked_curser = optional
+
+	onRemove()
+		if (QDELETED(src.owner))
+			return ..()
+		var/mob/living/L = src.owner
+		if (!isdead(L) && src.outputs_removal_msg)
+			boutput(L, SPAN_NOTICE(src.removal_msg))
+		src.linked_curser = null
+		..()
+
+	proc/get_mult(timePassed)
+		return timePassed / LIFE_PROCESS_TICK_SPACING
+
+	blood
+		id = "art_blood_curse"
+		name = "Blood Curse"
+		duration = null
+		extra_desc = "Your blood is being drained. The artifact requires 600u of human blood, or your drained body, no matter the cost. Figure out how to supply it before you die."
+		removal_msg = "Your blood curse has been lifted!"
+		var/blood_to_collect = 600
+
+		onAdd(optional)
+			..()
+			var/mob/living/carbon/human/H = src.owner
+			H.regens_blood = FALSE // curse steals all your blood
+
+		onUpdate(timePassed)
+			..()
+			var/mob/living/carbon/human/H = src.owner
+			var/mult = src.get_mult(timePassed)
+			H.blood_volume -= 3 * mult
+			if (H.bleeding <= 1) // mostly enabled to show bleed indicator
+				H.bleeding = 1
+			src.blood_to_collect -= 3 * mult
+			if (probmult(7))
+				boutput(H, SPAN_ALERT(pick("You see things", "You have thoughts about blood", "You can feel an Eldritch presence", "You can feel your blood",
+					"You get the sense something is stealing from you", "Something doesn't feel right", "The artifact hungers", "You see visions of Eldritch artifacts",
+					"You're reminded of your blood curse", "You have a pact to fulfill", "You're going to die unless blood is given", "Blood is required")))
+
+			if (src.blood_to_collect <= 0)
+				src.linked_curser.lift_curse(TRUE)
+			else if (H.blood_volume <= 0 || isdead(H))
+				H.visible_message(SPAN_ALERT("[H] spontaneously implodes!!! <b>HOLY FUCK!!</b>"), SPAN_ALERT("<b>Ohhhh shit</b>"))
+				for (var/i in 1 to rand(3, 4))
+					var/obj/decal/cleanable/blood_splat = make_cleanable(/obj/decal/cleanable/blood/splatter, get_turf(H))
+					blood_splat.streak_cleanable(pick(cardinal), full_streak = prob(25), dist_upper = rand(4, 6))
+				playsound(H, 'sound/impact_sounds/Slimy_Splat_2_Short.ogg', 40, TRUE)
+				H.implode(TRUE)
+				src.linked_curser.lift_curse_specific(FALSE, H)
+
+		onRemove()
+			var/mob/living/carbon/human/H = src.owner
+			if (!QDELETED(H))
+				H.regens_blood = TRUE
```

Comment:
Should migrate to an atomprop - otherwise, if you get 2 blood curses (somehow), clearing one will reenable blood regen even though you're still under a curse